### PR TITLE
[8.0][ADD] Migration scripts for sale_sourced_by_line

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Added migration scripts and analysis file.

### CODE REVIEW

File: sale_sourced_by_line/migrations/8.0.1.1.0/post-migration.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+
+def populate_order_line_location_id(env):
+    env.cr.execute(
+        """
+        SELECT id, %s
+        FROM sale_order_line
+        """ % openupgrade.get_legacy_name('location_id')
+    )
+    locations = env.cr.fetchall()
```
Comment (aheficent) [inline sale_sourced_by_line/migrations/8.0.1.1.0/post-migration.py]: Why locations instead of so_lines?

Comment (mreficent) [reply sale_sourced_by_line/migrations/8.0.1.1.0/post-migration.py]: It's just a name. That fetch gives a list of two fields: one is the so_line.id and the other is the so_line.location_id.

Comment (aheficent) [reply sale_sourced_by_line/migrations/8.0.1.1.0/post-migration.py]: You're right. It's ok to me.

**Comment by MiquelRForgeFlow** [comment]
ping: @pedrobaeza

**Comment by pedrobaeza** [comment]
Sorry, I don't know this module, so I can't say.

File: sale_sourced_by_line/migrations/8.0.1.1.0/post-migration.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from openupgradelib import openupgrade
+
+
+def populate_order_line_location_id(env):
+    env.cr.execute(
+        """
+        SELECT id, %s
+        FROM sale_order_line
+        """ % openupgrade.get_legacy_name('location_id')
```
Comment (LoisRForgeFlow) [inline sale_sourced_by_line/migrations/8.0.1.1.0/post-migration.py]: ~sale_order_line doesn't have location_id AFAIK~

edit: sorry the module added it in v8

File: sale_sourced_by_line/migrations/8.0.1.1.0/post-migration.py (hunk 1 above)
Comment (LoisRForgeFlow) [inline sale_sourced_by_line/migrations/8.0.1.1.0/post-migration.py]: name is misleading

File: sale_sourced_by_line/migrations/8.0.1.1.0/post-migration.py (hunk 2 above)
Comment (MiquelRForgeFlow) [reply sale_sourced_by_line/migrations/8.0.1.1.0/post-migration.py]: in v7 :ghost: :rofl:

Comment (lreficent) [reply sale_sourced_by_line/migrations/8.0.1.1.0/post-migration.py]: yes

Comment (LoisRForgeFlow) [reply sale_sourced_by_line/migrations/8.0.1.1.0/post-migration.py]: yes...

giphy

**Review by LoisRForgeFlow — APPROVED** [review]
LGTM now :+1:

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: LGTM

**Comment by JordiBForgeFlow** [comment]
@mreficent Please optimize..

**Comment by MiquelRForgeFlow** [comment]
@jbeficent Optimized

**Comment by MiquelRForgeFlow** [comment]
ping: @StefanRijnhart

**Review by StefanRijnhart — APPROVED** [review]

**Pull request merged**